runtime/README.md

```diff
@@ -1226,129 +1237,239 @@ Command Code: `0x434D_4B54` ("CMKT")
+| ----------- | -------- | ----------------------------------------|
+| chksum      | u32      |                                         |
+| fips_status | u32      | FIPS approved or an error               |
+| OKM CMK     | CMK      | CMK that stores the output key material |
 
 
-### CM_AES_GCM_ENCRYPT
+### CM_AES_GCM_ENCRYPT_INIT
 
 Currently only supports AES-256-GCM with a random 96-bit IV.
 
-If the key material is larger than 256 bits, then it will be truncated before use.
+The CMK must have been created for AES usage.
 
-Command Code: `0x434D_4745` ("CMGE")
+Command Code: `0x434D_4749` ("CMGI")
 
-*Table: `CM_AES_GCM_ENCRYPT` input arguments*
+*Table: `CM_AES_GCM_ENCRYPT_INIT` input arguments*
 | **Name**       | **Type** | **Description**                          |
 | -------------- | -------- | ---------------------------------------- |
 | chksum         | u32      |                                          |
-| flags          | u32      | Bit 0 = this is the final message        |
-| key CMID       | u8[32]   | CMID of the key to use                   |
-| tag size       | u32      | Number of bytes to return for tag.       |
-|                |          | Can be 0, 1, ..., 16                     |
-| context size   | u32      | Size of the context                      |
-| context        | u8[...]  |                                          |
+| CMK            | CMK  | CMK of the key to use to encrypt         |
 | aad size       | u32      | Additional authenticated data size       |
-|                |          | Must be 0 if this is not the first block |
 | aad            | u8[...]  | Additional authenticated data            |
 | plaintext size | u32      |                                          |
 | plaintext      | u8[...]  | Data to encrypt                          |
 
-*Table: `CM_AES_GCM_ENCRYPT` output arguments*
-| **Name**       | **Type** | **Description**                           |
-| -------------- | -------- | ----------------------------------------- |
-| chksum         | u32      |                                           |
-| fips_status    | u32      | FIPS approved or an error                 |
-| context size   | u32      | SHALL be 0 if this is the final message   |
-| context        | u8[...]  |                                           |
-| cipertext size | u32      | MAY be 0 if this is not the final message |
-| ciphertext     | u8[...]  |                                           |
+*Table: `CM_AES_GCM_ENCRYPT_INIT` output arguments*
+| **Name**       | **Type** | **Description**           |
+| -------------- | -------- | ------------------------- |
+| chksum         | u32      |                           |
+| fips_status    | u32      | FIPS approved or an error |
+| context size   | u32      |                           |
+| context        | u8[...]  |                           |
+| iv             | u8[12]   |                           |
+| cipertext size | u32      | MAY be 0                  |
```

Comment:
> MAY be 0

We would expect this to always equal the `plaintext size` from the input args, right? Ditto for other ciphertext/plaintext size return args below.

Reply:
Yep. I was probably thinking of the SHA cases, where you have to buffer a whole block, but this should always be equal to the plaintext size. I'll adjust this language.